daemon/src/model.rs

```diff
@@ -647,43 +647,41 @@ impl str::FromStr for FundingRate {
     }
 }
 
-/// Fee paid by the taker to the maker for opening a cfd.
-#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Serialize, Deserialize)]
+#[derive(thiserror::Error, Debug)]
+pub enum ConversionError {
+    #[error("Underflow")]
+    Underflow,
+    #[error("Overflow")]
+    Overflow,
+}
+
+/// Opening fee is always payed from taker to maker
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct OpeningFee {
-    amount_sat: u64,
+    #[serde(with = "bdk::bitcoin::util::amount::serde::as_sat")]
+    fee: Amount,
 }
 
 impl OpeningFee {
-    pub fn new(amount: Amount) -> Self {
-        Self {
-            amount_sat: amount.as_sat(),
-        }
+    pub fn new(fee: Amount) -> Self {
+        Self { fee }
     }
 
-    /// Value in satoshis for the taker
-    pub fn to_u64(&self) -> u64 {
-        self.amount_sat
-    }
+    fn into_signed_amount(self, role: Role) -> SignedAmount {
```

Comment:
I think we can simplify the code in settle etc if we don't have this indirection.

I'd be okay with just accessing the private fields of the other newtypes directly. We are in the same module so we can bundle all the invariants together.

Reply:
I agree. I already had the feeling it would be better to inline after being done, but was not sure about it. Will inline and keep the `OpeningFee` and `FundingFee` slim.